### PR TITLE
Unify approach of GPS location updates

### PR DESCRIPTION
**Pull request by KacperKluka**
I've changed our logic of starting and stopping location updates to work in the same way it works on iOS.
- The location updates are started when the first trackable is added to the Publisher
- The location updates are stopped when the last trackable is removed from the Publisher

### CODE REVIEW

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt [hunk 1]
```diff
@@ -226,7 +217,24 @@ constructor(
...
-                        createChannelForTrackableIfNotExisits(event.trackable, event.handler, state)
+                        ably.connect(event.trackable.id, state.presenceData) { result ->
+                            try {
+                                result.getOrThrow()
+                                try {
+                                    ably.subscribeForPresenceMessages(event.trackable.id) {
+                                        enqueue(PresenceMessageEvent(event.trackable, it))
+                                    }
+                                } catch (exception: ConnectionException) {
+                                    // TODO - what to do here? should we fail the whole process when subscribing for presence fails? or should it continue?
```
Comment (QuintinWillison) [inline publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt:228]: Do we have an explicit issue for this yet?

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt [hunk 2]
```diff
@@ -226,7 +217,24 @@ constructor(
...
+                                }
+                                ably.subscribeForChannelStateChange(event.trackable.id) {
+                                    enqueue(ChannelConnectionStateChangeEvent(it, event.trackable.id))
+                                }
+                                request(ConnectionForTrackableCreatedEvent(event.trackable, event.handler))
```
Comment (QuintinWillison) [inline publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt:233]: I've mentioned this on a previous PR, but I'm not sure that the `request` verb feels correct given the event name.

**Review by QuintinWillison — APPROVED** [review]

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt (hunk 1 above)
Comment (KacperKluka) [reply publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt:228]: Yes we do :wink: https://github.com/ably/ably-asset-tracking-android/issues/214

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt (hunk 2 above)
Comment (KacperKluka) [reply publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt:233]: Right, let's discuss it under that [PR comment](https://github.com/ably/ably-asset-tracking-android/pull/224#discussion_r602286727) :wink:

**Pull request merged**